File: python-leetcode/dynamic-programming/319.杨辉三角.py
from typing import List


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        res = []

        for i in range(numRows):
            # 每行 + 1  个数
            rows = [1] * (i + 1)

            for j in range(1, len(rows) - 1):
                rows[j] = res[i - 1][j - 1] + res[i - 1][j]

            res.append(rows)

        return res

File: go-leetcode/dynamic-programming/319.杨辉三角.go
package leetcode

func generate(numRows int) [][]int {
	res := [][]int{}

	for i := 0; i < numRows; i++ {
		// 每行 + 1  个数
		rows := make([]int, i+1)
		for idx := range rows {
			rows[idx] = 1
		}

		for j := 1; j < len(rows)-1; j++ {
			rows[j] = res[i-1][j-1] + res[i-1][j]
		}

		res = append(res, rows)
	}

	return res
}
